library/general/src/lib/ui/multi_messages_dialog.rb

```diff
@@ -0,0 +1,247 @@
+require "yast"
+require "ui/dialog"
+
+Yast.import "UI"
+Yast.import "Label"
+
+module UI
+  # This dialog receives several messages and allows the user to browse them
+  # using 'Next' and 'Back' buttons.
+  #
+  # Each message is an object that should respond to #title and #body methods.
+  class MultiMessagesDialog < ::UI::Dialog
+    include Yast::Logger
+
+    DEFAULT_MIN_HEIGHT = 20
+    DEFAULT_MIN_WIDTH = 60
+    TIMEOUT_STEP = 1000 # miliseconds
+
+    attr_reader :messages, :headline, :min_height, :min_width, :timeout
+
+    # Constructor
+    #
+    # @param headline   [String]  Dialog's headline
+    # @param messages   [Array]   Array of messages. Objects in the array should
+    #                             respond to #title and #body methods.
+    # @param min_height [Integer] Minimal dialog's height.
+    # @param min_width  [Integer] Minimal dialog's width.
+    def initialize(headline, messages, min_height: nil, min_width: nil, timeout: false)
+      super()
+      textdomain "yast"
+      @messages = messages
+      @position = 0
+      @headline = headline
+      @min_height = min_height || DEFAULT_MIN_HEIGHT
+      @min_width = min_width || DEFAULT_MIN_WIDTH
+      @timeout = timeout
+      @remaining_time = timeout
+    end
+
+    # Dialog's content
+    #
+    # Overrides UI::Dialog#dialog_content
+    #
+    # @return [Yast::Term] UI content for dialog
+    #
+    # @see UI::Dialog#dialog_content
+    def dialog_content
```

Comment:
NP: IMHO this and some other methods should be private (or protected). It should be only possible to create and run the dialog. The other methods are just implementation details and should be hidden from outside.